compiler/aarch64/codegen/ARM64Instruction.hpp

```diff
@@ -2669,30 +2599,12 @@ class ARM64SynchronizationInstruction : public TR::Instruction
     */
    virtual Kind getKind() { return IsSynchronization; }
 
-   /**
-    * @brief Gets source immediate
-    * @return source immediate
-    */
-   uint32_t getSourceImmediate() {return _sourceImmediate;}
-   
-   /**
-    * @brief Sets source immediate
-    * @param[in] si : immediate value
-    * @return source immediate
-    */
-   uint32_t setSourceImmediate(uint32_t si) {return (_sourceImmediate = si);}
-
-   void insertImmediateField(uint32_t *instruction)
+   virtual void insertImmediateField(uint32_t *instruction)
       {
-      TR_ASSERT(_sourceImmediate <= 0xF, "Immediate value exceeds 4 bits.");
-      *instruction |= ((_sourceImmediate & 0xF) << 8);
+      TR_ASSERT(constantIsUnsignedImm4(getSourceImmediate()), "Immediate value exceeds 4 bits.");
```

Comment:
This sounds like a fatal assert. If we cannot encode the immediate the program is not functionally correct.

Reply:
Agree.  Fixed.